src/System.Linq/src/System/Linq/Concat.cs

```diff
@@ -61,16 +60,15 @@ internal Concat2Iterator(IEnumerable<TSource> first, IEnumerable<TSource> second
 
             internal override ConcatIterator<TSource> Concat(IEnumerable<TSource> next)
             {
-                bool hasOnlyCollections = _first is ICollection<TSource> &&
-                                          _second is ICollection<TSource> &&
-                                          next is ICollection<TSource>;
+                bool hasOnlyCollections = next is ICollection<TSource> &&
+                                          _first is ICollection<TSource> &&
+                                          _second is ICollection<TSource>;
                 return new ConcatNIterator<TSource>(this, next, 2, hasOnlyCollections);
             }
 
             public override int GetCount(bool onlyIfCheap)
             {
-                int firstCount, secondCount;
-                if (!EnumerableHelpers.TryGetCount(_first, out firstCount))
+                if (!EnumerableHelpers.TryGetCount(_first, out int firstCount))
```

Comment:
NP, though personally I find the new style an improvement because it's clearly assigned at that point.